x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/Collectable.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rollup.v2;
+
+/**
+ * Base interface for classes that collect metric and label field values.
+ */
+interface Collectable<T> {
+    void collect(T value);
+}
```

Comment:
Do we actually use this interface anywhere? It's implemented by the FieldProducers but we never refer to anything as a Collectable so I think we can probably just fold it into AbstractFieldProducer